rs/bitcoin/checker/src/main.rs

```diff
@@ -421,8 +410,26 @@ impl FetchEnv for BtcCheckerCanisterEnv {
                 }
                 Ok(tx)
             }
-            Err((r, m)) if is_response_too_large(&r, &m) => Err(HttpGetTxError::ResponseTooLarge),
+            Err((r, m)) if is_response_too_large(&r, &m) => {
+                if max_response_bytes >= RETRY_MAX_RESPONSE_BYTES {
+                    STATS.with(|s| {
+                        let mut stat = s.borrow_mut();
+                        *stat
+                            .https_outcall_status
+                            .entry((provider.name(), "ResponseTooLarge".to_string()))
+                            .or_default() += 1;
+                    });
+                }
+                Err(HttpGetTxError::ResponseTooLarge)
+            }
             Err((r, m)) => {
+                STATS.with(|s| {
+                    let mut stat = s.borrow_mut();
+                    *stat
+                        .https_outcall_status
+                        .entry((provider.name(), format!("{:?}", r)))
```

Comment:
Maybe it would be better to have an enum for the values in `https_outcall_status` rather than a string for the following reasons

1. Easier to find out when a concrete variant, such as `ResponseTooLarge` is used.
2. Not dependent on debug representation of `RejectionCode`, which is not guaranteed to be stable.

We could have something like this, WDYT?
```rust
enum HttpOutcallStatus {
  ResponseTooLarge,
  IcError(RejectionCode)
  HttpStatusCode(u16)
}
```

Reply:
Good idea! Introduced an enum for the status type. Please have a look. Thanks!

Reply:
Thanks for the changes!